File: typescript/src/resources/tw.ts
/** X/Twitter data resources (`/x/*`). */

import { Transport } from '../runtime/transport';

export class Tw {
  constructor(private transport: Transport) {}

  posts(opts: { userId: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { userId, ...rest } = opts;
    return this.transport.request('POST', '/x/posts', { json: { user_id: userId, ...rest } });
  }

  users(opts: { username: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    return this.transport.request('POST', '/x/users', { json: opts });
  }

  retweets(opts: { keyword: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    return this.transport.request('POST', '/x/retweets', { json: opts });
  }
}

File: python/src/acedatacloud/resources/tw.py
"""X/Twitter data resources (``/x/*``)."""

from __future__ import annotations

from typing import Any


class Tw:
    """Synchronous X/Twitter client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def posts(self, *, user_id: str, **kwargs: Any) -> dict[str, Any]:
        return self._transport.request("POST", "/x/posts", json={"user_id": user_id, **kwargs})

    def users(self, *, username: str, **kwargs: Any) -> dict[str, Any]:
        return self._transport.request("POST", "/x/users", json={"username": username, **kwargs})

    def retweets(self, *, keyword: str, **kwargs: Any) -> dict[str, Any]:
        return self._transport.request("POST", "/x/retweets", json={"keyword": keyword, **kwargs})


class AsyncTw:
    """Async X/Twitter client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def posts(self, *, user_id: str, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request("POST", "/x/posts", json={"user_id": user_id, **kwargs})

    async def users(self, *, username: str, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request("POST", "/x/users", json={"username": username, **kwargs})

    async def retweets(self, *, keyword: str, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request("POST", "/x/retweets", json={"keyword": keyword, **kwargs})
